crates/service-client/src/http.rs

```diff
@@ -20,26 +20,32 @@ use hyper::http::HeaderValue;
 use hyper::{Body, HeaderMap, Method, Request, Response, Uri, Version};
 use hyper_rustls::HttpsConnector;
 use restate_types::config::HttpOptions;
+use std::error::Error;
 use std::fmt::Debug;
 use std::future;
 use std::future::Future;
-
 type Connector = ProxyConnector<HttpsConnector<HttpConnector>>;
 
 #[derive(Clone, Debug)]
 pub struct HttpClient {
-    client: hyper::Client<Connector, Body>,
+    alpn_client: hyper::Client<Connector, Body>,
+    h2_client: hyper::Client<Connector, Body>,
```

Comment:
Maybe add a quick comment explaining why we need two clients instead of `alpn` only. Primarily for my future self.